es6-operators/main.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-vars */
+const v1 = { value: 'Something' };
+const v2 = '';
+const v3 = undefined;
+const v4 = [3, 5, 7, 9];
+
+/* TODO:
+ * Demonstrate how to execute an expression only if a variable is truthy.
+ * In one expression, log "v1 is truthy" to the console only if `v1` is truthy.
+ * In one expression, log "v2 is truthy" to the console only if `v2` is truthy.
+ * Expected: v1 message logs, v2 message does not.
+ */
+v1 ? console.log('v1 is truthy') : null;
+v2 ? console.log('v2 is truthy') : null;
+/* TODO:
+ * Demonstrate how to assign a "default value" to a variable.
+ * In one expression, assign `'default-value'` to the variable `config1` if v1 is falsy.
+ * In one expression, assign `'default-value'` to the variable `config2` if v2 is falsy.
+ * Log both `config1` and `config2`.
+ * Expected: `config1 = { value: 'Something' }`, `config2 = 'default-value'`.
+ */
+
+const config1 = v1 || 'default-value';
+const config2 = v2 || 'default-value';
+console.log('config1: ', config1);
+console.log('config2: ', config2);
+/* TODO:
+ * Demonstrate how to assign a "default value" to a variable only if the variable is `undefined`.
+ * In one expression, assign `'default-value'` to the variable `cfg1` if v1 is undefined.
+ * In one expression, assign `'default-value'` to the variable `cfg2` if v2 is undefined.
+ * In one expression, assign `'default-value'` to the variable `cfg3` if v3 is undefined.
+ * Log `cfg1`, `cfg2`, and `cfg3`.
+ * Expected: `config1 = { value: 'Something' }`, `config2 = ''`, `config3 = 'default-value'`.
+ */
+const cfg1 = v1 !== undefined ? v1 : 'default-value';
+const cfg2 = v2 !== undefined ? v2 : '';
+const cfg3 = v3 !== undefined ? v3 : 'default-value';
+
+console.log('cfg1: ', cfg1);
+console.log('cfg2: ', cfg2);
+console.log('cfg3: ', cfg3);
+
+/* TODO:
+ * Use the shorthand if/else, which is especially useful for variable assignments.
+ * In one expression, assign `truthy` to the variable `tern1` if v1 is truthy
+ * and assign `falsy` to the same variable if v1 is falsy.
+ * Do the same for the variable `tern2` and the variable v2.
+ * Log `tern1` and `tern2`.
+ * Expected: `tern1 = 'truthy'`, `tern2 = 'falsy'`.
+ */
+const tern1 = v1 ? 'truthy' : 'falsy';
+const tern2 = v2 ? 'truthy' : 'falsy';
+
+console.log('tern1: ', tern1);
+console.log('tern2: ', tern2);
+
+/* TODO:
+ * Use the shorthand for "if defined".
+ * In one expression, assign the `value` property of variable v1 to the variable
+ * `oc1`, but only if it is defined.
+ * Do the same for the variable `oc2` and the variable v2.
+ * Do the same for the variable `oc3` and the variable v3.
+ * Log `oc1`, `oc2`, and `oc3`.
+ * Expected: `oc1 = 'Something'`, `oc2 = undefined`, `oc3 = undefined`.
+ */
+const oc1 = v1?.value || undefined;
+const oc2 = v2?.value || undefined;
+const oc3 = v3?.value || undefined;
```

Comment:
Optional chaining already returns `undefined` for us if the value doesn't exist on the object! That's the syntatic sugar of optional chaining!  
```
const oc1 = v1?.value;
const oc2 = v2?.value;
const oc3 = v3?.value;
```